cmd/template/nodepool/runner.go

```diff
@@ -85,12 +91,16 @@ func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) err
 		if r.flag.Provider == key.ProviderAWS {
 			config.Namespace = r.flag.ClusterNamespace
 		}
-	}
 
-	commonConfig := commonconfig.New(r.flag.config)
-	c, err := commonConfig.GetClient(r.logger)
-	if err != nil {
-		return microerror.Mask(err)
+		wc, err := getWorkloadCluster(ctx, c, config.Namespace, config.ClusterName, r.flag.Provider)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		if len(config.ReleaseVersion) < 1 {
+			config.ReleaseVersion = wc.Cluster.Labels[label.ReleaseVersion]
+		} else {
+			config.ReleaseVersion = strings.TrimLeft(config.ReleaseVersion, "v")
```

Comment:
I'm not even sure we need an `else` case here. I can not think of any situation where we would want the nodepool to be a different version than the cluster.

Reply:
The `else` branch is used when the user provides the `--release` flag, and the value isn't defaulted.

If we want to remove the flag, it should probably be deprecated first for some time